app/src/lib/stores/repositories-store.ts

```diff
@@ -334,4 +335,62 @@ export class RepositoriesStore extends BaseStore {
       repository.missing
     )
   }
+
+  public async updateLastPruneDate(repository: Repository): Promise<void> {
+    const repoID = repository.id
+    if (!repoID) {
+      return fatalError(
+        '`updateLastPruneDate` can only update the last prune date for a repository which has been added to the database.'
+      )
+    }
+
+    const githubRepo = repository.gitHubRepository
+    if (githubRepo === null) {
+      return fatalError(
+        `'updateLastPruneDate' can only update GitHub repositories`
+      )
+    }
+
+    const gitHubRepositoryID = githubRepo.dbID
+    if (gitHubRepositoryID === null) {
+      return fatalError(
+        `'updateLastPruneDate' can only update GitHub repositories with a valid ID: received ID of ${gitHubRepositoryID}`
+      )
+    }
+
+    await this.db.gitHubRepositories.update(gitHubRepositoryID, {
+      lastPruneDate: Date.now(),
+    })
+
+    this.emitUpdate()
+  }
+
+  public async getLastPruneDate(
+    repository: Repository
+  ): Promise<number | null> {
+    const repoID = repository.id
+    if (!repoID) {
+      return fatalError(
+        '`getLastPruneDate` can only retrieve the last prune date for a repositories that have been stored in the database.'
+      )
+    }
+
+    const githubRepo = repository.gitHubRepository
+    if (githubRepo === null) {
+      return fatalError(
+        `'getLastPruneDate' can only retrieve the last prune date for GitHub repositories.`
+      )
+    }
+
+    const gitHubRepositoryID = githubRepo.dbID
+    if (gitHubRepositoryID === null) {
+      return fatalError(
+        `'getLastPruneDate' can only retrieve the last prune date for GitHub repositories that have been stored in the database.`
+      )
+    }
+
+    const record = await this.db.gitHubRepositories.get(gitHubRepositoryID)
+
+    return record!.lastPruneDate
```

Comment:
I really don't think we should use `!` except for in tests. If we're certain that this can't ever be `undefined` (which I don't think we can be) we should check and throw if it is.